### PR TITLE
feat: return sharees when returning a workflow

### PR DESCRIPTION
**Pull request by krynble**

### CODE REVIEW

**Review by freyamade — APPROVED** [review]
All looks good to me!

**Review by valya — COMMENTED** [review]
It mostly looks good. My one concern is that it's overriding an existing route. This route probably isn't going to change any time soon but it does worry me that they could fall out of sync. Not sure if there's really a way to address it at the moment though.

File: packages/cli/src/workflows/workflows.controller.ee.ts [hunk 1]
```diff
@@ -58,3 +58,37 @@ EEWorkflowController.put('/:workflowId/share', async (req: WorkflowRequest.Share
 
 	return res.status(200).send();
 });
+
+EEWorkflowController.get(
+	'/:id',
```
Comment (netroy) [inline packages/cli/src/workflows/workflows.controller.ee.ts:63]: you could do `/:id(\\d+)` here, and then remove the check for `new` as well as the `isNaN` check.

File: packages/cli/src/workflows/workflows.controller.ee.ts [hunk 2]
```diff
@@ -58,3 +58,37 @@ EEWorkflowController.put('/:workflowId/share', async (req: WorkflowRequest.Share
...
+			throw new ResponseHelper.ResponseError(
+				`Workflow with ID "${workflowId}" could not be found.`,
+				undefined,
+				404,
+			);
+		}
+
+		const userSharing = workflow.shared?.find((shared) => shared.user.id === req.user.id);
+
+		if (!userSharing && req.user.globalRole.name !== 'owner') {
```
Comment (netroy) [inline packages/cli/src/workflows/workflows.controller.ee.ts:87]: I think we should extract the `globalRole` check into a middleware. This won't be the last place we need this check.

**Comment by janober** [comment]
Got released with `n8n@0.198.0`

**Pull request merged**